### PR TITLE
[MIG][9.0] product_multi_image to v9

### PR DESCRIPTION
**Pull request by lasley**
[MIG] product_multi_image: Upgrade to v9  …
* Upgrade product_multi_image to v9

This should supersede #161 due to inactivity.

This is dependent on merge of:
- [x] OCA/server-tools#628

### CODE REVIEW

File: product_multi_image/__openerp__.py [hunk 1]
```diff
@@ -5,13 +5,13 @@
 
 {
     "name": "Multiple Images in Products",
-    "version": "8.0.2.0.0",
-    "author": "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
```
Comment (lasley) [inline product_multi_image/__openerp__.py]: @pedrobaeza - I took the liberty of removing your old company and changing the URI to Tecnativa. Let me know if I was incorrect with these changes

Comment (pedrobaeza) [reply product_multi_image/__openerp__.py]: Of course, thanks

**Comment by lasley** [comment]
I'll get the Travis resolved sometime today

**Comment by lasley** [comment]
Bah well this error is a real scumbag, and is most definitely legit (good job with your tests @pedrobaeza).

```
>>> template = self.env['product.template'].create({
        'name': 'Test 2',
        'image_ids': [(0, 0, {
            'storage': 'db',
            'name': 'Image 1',
            'file_db_store': self.transparent_image,
            'owner_model': 'product.template',
        })],
    })
>>> len(template.image_ids)
1
>>> template.write({
        'attribute_line_ids': [
            (0, 0, {
                'attribute_id': self.attribute.id,
                'value_ids': [(6, 0, (self.value_1 + self.value_2).ids)],
            })],
    })
>>> len(template.image_ids)
0
```

I'm playing around with it in `pdb` at the moment, just wanted to provide an update because I said this would be done yesterday 😆

**Comment by simahawk** [comment]
@lasley yep, I debugged it a couple of days ago and I was stuck here, when you [delete the product](https://github.com/laslabs/product-attribute/blob/5e5f2bf329f9bd99c360621fc220f1b7de7be795/product_multi_image/tests/test_product_multi_image.py#L134) the override of [unlink](https://github.com/laslabs/product-attribute/blob/5e5f2bf329f9bd99c360621fc220f1b7de7be795/product_multi_image/models/product_product.py#L98) deletes ALL the images.
I've not that much background on the whole module so after a while a gave up, willing to come back at some point :)
Let me know if I can help w/ further testing.

**Comment by pedrobaeza** [comment]
I can take a look also, but not before the end of this week.

**Comment by lasley** [comment]
Yeah it seems that the variants aren't getting appended into the `base_multi_image.image.product_variant_ids` upon initial creation, so I'm thinking the issue is actually in `_compute_image_ids`. 

```
(Pdb) template.image_ids
base_multi_image.image(517,)
(Pdb) template.image_ids.product_variant_ids
product.product()
```

I think it may have something to do with the fact that [`product.template` now defines the image fields using new API](https://github.com/odoo/odoo/blob/9.0/addons/product/product.py#L538), but the `product.product` still [uses old](https://github.com/odoo/odoo/blob/9.0/addons/product/product.py#L972). Maybe something with the API wire crossing similar to what is described in https://github.com/odoo/odoo/issues/10799

**Comment by lasley** [comment]
We can scratch API wire crossing off the list of possible issues, #201 😦

**Comment by lasley** [comment]
This is solved in OCA/server-tools#628 & should be ready for review after merge

**Comment by pedrobaeza** [comment]
Rebase your branch

**Comment by pedrobaeza** [comment]
Actually only a rebuild is needed (I have launched it)

**Review by pedrobaeza — APPROVED** [review]
It would be good to complete tests testing product_variant_count computed field and uninstallation hook

**Comment by lasley** [comment]
@pedrobaeza - I added one for the pre_init as well, Coverage was a free/false green there because it's a guaranteed run. We'll see if they pass, I didn't actually run in local TBH 😉

**Comment by pedrobaeza** [comment]
I'm afraid there's no luck.

**Comment by lasley** [comment]
Even more disconcerting is the fact that Runbot is 🍏

**Comment by pedrobaeza** [comment]
:clap: :clap: :clap:

**Review by sebalix — APPROVED** [review]
Wonderful

**Comment by pedrobaeza** [comment]
@lasley Please squash your commits and I'll merge.

**Comment by lasley** [comment]
Squashed up! I'll knock out the v10 in the next few days

**Comment by simahawk** [comment]
yay! thanks @lasley !

**Pull request merged**